data_collection/collecting_articles.py

```diff
@@ -0,0 +1,120 @@
+        except :
+            periodic["issues"] = "unknown"
+        results.append(periodic)
+    return results
+
+def get_issues_reference_by_name(title,yearStart,yearEnd,ocrQuality=0.6):
+    """find the references of all issues of a newspaper with a given name in a given time period
+    
+    Arguments:
+        title {string} -- the name of the newspaper to search issues for
+        yearStart {int} -- the beginning of the period to search issues for
+        yearEnd {int} -- the end of the period to search issues for
+    
+    Keyword Arguments:
+        ocrQuality {float} -- minimal ocr quality accepted (default: {0.6})
+    
+    Returns:
+        list -- a list of all the references of the issues from the given newspaper
+    """
+    cqlQuery = f"dc.title any '{title}' and dc.type any fascicule"
+    r = requests.get(f"http://gallica.bnf.fr/SRU?version=1.2&maximumRecords=1&operation=searchRetrieve&query={cqlQuery}")
+    strContent = r.content.decode("utf-8")
+    xmlContent = ElementTree.fromstring(strContent)
+    newspaper_ref = xmlContent.find("srw:records/srw:record/srw:recordData/oai_dc:dc/dc:identifier",ns).text.split("/")[5]
+    return get_issues_reference_by_newspaper_ref(newspaper_ref,yearStart,yearEnd,ocrQuality)
+
+def get_issues_reference_by_newspaper_ref(ref,yearStart,yearEnd,ocrQuality=0.6):
+    """find the references of all issues of a newspaper with a given reference in a given time period
+    
+    Arguments:
+        ref {string} -- ark reference (official gallica reference) of the newspaper
+        yearStart {int} -- beginning of the time period to search issues for
+        yearEnd {int} -- end of the time period to search issues for
+    
+    Keyword Arguments:
+        ocrQuality {float} -- the minimal ocr quality accepted (default: {0.6})
+    
+    Returns:
+        list -- a list of all the references of the issues from the given newspaper
+    """
+    refs = []
+    for year in tqdm(range(yearStart,yearEnd+1)):
+        r = requests.get(f"https://gallica.bnf.fr/services/Issues?ark=ark:/12148/{ref}/date&date={year}")
+        strIssues = r.content.decode("utf-8")
+        try :
+            xmlIssues = ElementTree.fromstring(strIssues)
+            for issue in xmlIssues:
+                refs.append(issue.attrib["ark"])
+        except :
+            logging.info(f"no issues for year {year}")
+    return refs
+
+def get_issue_by_ref(ref):
+    """retrieve the content of an issue given its reference
+    
+    Arguments:
+        ref {string} -- ark reference of an the issue
+    
+    Returns:
+        dict -- an issue dictionnary
+        dict-keys :
+            newspaper {string}: name of the newspaper the issue comes from
+            date {string}: date of the issue
+            raw_text {list of strings}: all text within the issue
+            url {string}: the url the issue can be found at
+
+    """
+    r = requests.get(f"https://gallica.bnf.fr/ark:/12148/{ref}.texteBrut")
```

Comment:
Avoid hard coding stuff (url, strings, dates, int) if you use them multiple times. You can set attributes or create configuration files